fltk/federator.py

```diff
@@ -78,6 +78,14 @@ def __init__(self, client_id_triple, num_epochs = 3, config=None):
         self.create_clients(client_id_triple)
         self.config.init_logger(logging)
 
+        logging.info("Creating test client")
+        copy_sampler = config.data_sampler
+        config.data_sampler = "uniform"
+        self.test_data = Client("test", None, 1, 2, config)
+        self.test_data.init_dataloader()
+        config.data_sampler = copy_sampler
```

Comment:
Do we test on the whole (test) dataset or on only the part allocated based on the rank, as the naive distributed sampler does?

Reply:
The federator tests on the whole test dataset. It is actually based on rank, but the world size for this 'client' is set to 2, so the result is the same. Using a client here made it easy to have a global test dataset.